python/mp/effects/effect.py

```diff
@@ -77,24 +79,33 @@ def get_name(self):
         """Returns the name of the Effect."""
         return self.name
 
-    @abc.abstractmethod
-    def next(self):
+    def supernext(self):
         """
         Invoked for every mainloop cycle.
-        This method _must_ be invoked by every Effect's next() method.
+
+        Having the rosary call this method, and having this method call
+        the actual `next()` method lets me make generically implement `delay`
+        without forcing every effect to re-implement it.
         """
-        self.color.next()
-
-        #print("DURATION: {}, TIME: {}".format(self.duration, self.time))
-        if self.duration is not None and self.time >= self.duration:
-            print("I MUST GO NOW MY PEOPLE NEED ME")
-            # Is this redundant?
-            # NOTE: Figure out who's responsible for this: rosary? effect?
-            #self.unregister_with_dispatcher()
+
+        # In refactoring triggers, I wanted an alternate way to "script"
+        # sequences - if delay is passed, don't start "nexting" until it's over
+        if self.time > self.delay:
```

Comment:
We can quibble about wasting CPU later, but this was my workaround for losing the "scripting." that I had when I did `set_effect_sequence.`

So now all effects are created at the same time, and just told how long to wait before starting. This lets me do that.